### ISSUE TITLE
Make Sablon use live XML instead of strings until files are written

### ISSUE THREAD

**Issue by stadelmanma** [stadelmanma/sablon#15]
Using live XML instead of strings when working with sablon is the first step towards a formal document object model and will allow a hige degree of flexibility because references to give portions of the document (i.e. bookmarks or footnote refs) can be maintained. This prevents the need for an "ast_to_docx" hook to do things like update my footnote references. This conversion will take some effort but I think it is worthwhile especially when we want to venture into other territories like adding media, docs partials, etc.

Basically instead of having the `to_docx` method return a string it would return an XML node or NodeSet that would be injected into the document at the proper location.

**Comment by stadelmanma** [comment]
Something to note is that the static XML generated by HTML conversion is actually re-parsed as XML prior to insertion. So this actually removes the intermediate step of saving to a string and then reparsing that string into live XML.

**Comment by stadelmanma** [comment]
The only real reason this was desired is to fix my broken model for adding footnotes to a document. A "live DOM" is a good idea for several reasons but this is not the step to implementing it.
